### DISCUSSION TITLE
Error occurs when running pip install fastapi (conda)

### DISCUSSION THREAD

**Discussion by vulcanidic95** [fastapi/fastapi#11167] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
$ conda create -n test_env python=3.11
$ conda activate test_env
$ (test_env) pip install fastapi
```

### Description

$ pip install fastapi
Looking in indexes: https://pypi.org/simple, https://test.pypi.org/simple/
Collecting fastapi
  Downloading https://test-files.pythonhosted.org/packages/fc/3f/1418252ee07b6bc3d1c8da2c5a2b24fd98200c1e534ff517f5006b9ba7bd/FASTAPI-1.0.tar.gz (2.5 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-gamnto8v/fastapi_ec0452af9f874e6e9c6f195023f9ed20/setup.py", line 3, in <module>
          with open('DESCRIPTION.txt') as file:
               ^^^^^^^^^^^^^^^^^^^^^^^
      FileNotFoundError: [Errno 2] No such file or directory: 'DESCRIPTION.txt'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

### Operating System

Linux

### Operating System Details

Ubuntu 20.04 & anaconda 23.7.4

### FastAPI Version

install error

### Pydantic Version

install error

### Python Version

3.11.7

**Comment by GodMoonGoodman** [comment]
I tried install fastapi in Docker container (ubuntu:20.04), but didn't get same result.

```
$ docker run --rm -it ubuntu:20.04
```

In container,
```
$ apt-get update
$ apt-get install wget build-essential checkinstall
$ apt-get install libreadline-gplv2-dev libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev
$ wget https://www.python.org/ftp/python/3.11.7/Python-3.11.7.tgz
$ tar -xvzf Python-3.11.7.tgz
$ cd Python-3.11.7.tgz
$ ./configure
$ make altinstall
```

```
$ cat /etc/os-release

NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

```
$ python3.11 -V

Python 3.11.7
```

```
$ pip install fastapi

Collecting fastapi
  Downloading fastapi-0.109.2-py3-none-any.whl.metadata (25 kB)
Collecting pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4 (from fastapi)
  Downloading pydantic-2.6.1-py3-none-any.whl.metadata (83 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.5/83.5 kB 9.4 MB/s eta 0:00:00
Collecting starlette<0.37.0,>=0.36.3 (from fastapi)
  Downloading starlette-0.36.3-py3-none-any.whl.metadata (5.9 kB)
Collecting typing-extensions>=4.8.0 (from fastapi)
  Downloading typing_extensions-4.9.0-py3-none-any.whl.metadata (3.0 kB)
Collecting annotated-types>=0.4.0 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi)
  Downloading annotated_types-0.6.0-py3-none-any.whl.metadata (12 kB)
Collecting pydantic-core==2.16.2 (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi)
  Downloading pydantic_core-2.16.2-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl.metadata (6.5 kB)
Collecting anyio<5,>=3.4.0 (from starlette<0.37.0,>=0.36.3->fastapi)
  Downloading anyio-4.3.0-py3-none-any.whl.metadata (4.6 kB)
Collecting idna>=2.8 (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi)
  Downloading idna-3.6-py3-none-any.whl.metadata (9.9 kB)
Collecting sniffio>=1.1 (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi)
  Downloading sniffio-1.3.0-py3-none-any.whl (10 kB)
Downloading fastapi-0.109.2-py3-none-any.whl (92 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 92.1/92.1 kB 9.7 MB/s eta 0:00:00
Downloading pydantic-2.6.1-py3-none-any.whl (394 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 394.8/394.8 kB 10.3 MB/s eta 0:00:00
Downloading pydantic_core-2.16.2-cp311-cp311-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (1.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB 11.4 MB/s eta 0:00:00
Downloading starlette-0.36.3-py3-none-any.whl (71 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 71.5/71.5 kB 11.9 MB/s eta 0:00:00
Downloading typing_extensions-4.9.0-py3-none-any.whl (32 kB)
Downloading annotated_types-0.6.0-py3-none-any.whl (12 kB)
Downloading anyio-4.3.0-py3-none-any.whl (85 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.6/85.6 kB 10.0 MB/s eta 0:00:00
Downloading idna-3.6-py3-none-any.whl (61 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.6/61.6 kB 8.1 MB/s eta 0:00:00
Installing collected packages: typing-extensions, sniffio, idna, annotated-types, pydantic-core, anyio, starlette, pydantic, fastapi
Successfully installed annotated-types-0.6.0 anyio-4.3.0 fastapi-0.109.2 idna-3.6 pydantic-2.6.1 pydantic-core-2.16.2 sniffio-1.3.0 starlette-0.36.3 typing-extensions-4.9.0
```

My opinion is reinstall python & pip.

**Comment by vulcanidic95** [comment]
The commands below work well.
$ pip install fastapi==0.1.16
$ pip install fastapi==0.109.2
$ pip install fastapi==0.109.1
$ pip install fastapi==0.109.0
$ pip install fastapi==0.108.0

The command below does not work.
$ pip install fastapi

**Comment by GodMoonGoodman** [comment]
try ```pip install fastapi -i https://pypi.org/simple```
not
try ```pip install fastapi -i https://test.pypi.org/simple```

**Reply by vulcanidic95** [reply]
same problems

> pip install fastapi -i https://pypi.org/simple
Looking in indexes: https://pypi.org/simple, https://test.pypi.org/simple/
Collecting fastapi
  Using cached https://test-files.pythonhosted.org/packages/fc/3f/1418252ee07b6bc3d1c8da2c5a2b24fd98200c1e534ff517f5006b9ba7bd/FASTAPI-1.0.tar.gz (2.5 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [7 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\dell\AppData\Local\Temp\pip-install-9ruzvayj\fastapi_d594793381fa49f9a08880d193774220\setup.py", line 3, in <module>
          with open('DESCRIPTION.txt') as file:
               ^^^^^^^^^^^^^^^^^^^^^^^
      FileNotFoundError: [Errno 2] No such file or directory: 'DESCRIPTION.txt'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

**Comment by YuriiMotov** [comment]
Is this still relevant?
Let's close this for now to keep the discussions clean and organized.
If you still need help, feel free to re-open this discussion.
